crates/polars-core/src/frame/mod.rs

```diff
@@ -2557,7 +2557,11 @@ impl DataFrame {
     pub fn mean_horizontal(&self, null_strategy: NullStrategy) -> PolarsResult<Option<Series>> {
         match self.columns.len() {
             0 => Ok(None),
-            1 => Ok(Some(self.columns[0].clone())),
+            1 => Ok(Some(if self.columns[0].dtype() == &DataType::Boolean {
```

Comment:
Thanks @nameexhaustion, I'll keep this as a draft until I can verify the other dtypes.

Reply:
I'm going to leave `Time` and `Date` dtypes to pass-through as `Time` and `Date` right now, even though the `Series` returns a float, because I view these implementations as errors, and there are open PRs to fix them already (#14471 and #13492).